src/explorer/weights/WeightSlider.js

```diff
@@ -2,30 +2,61 @@
 
 import React from "react";
 
-export const WEIGHT_SLIDER_MIN = -5;
-export const WEIGHT_SLIDER_MAX = 5;
+/**
+ * The Weight is the number that the WeightSlider actually provides to the
+ * instantiating component. The Weight varies in the range
+ * [0, 2**MAX_SLIDER_VALUE].
+ */
+export type Weight = number;
+
+/**
+ * The SliderPosition is the current position of the [range input] that the
+ * WeightSlider instantiates. It varies in the range [MIN_SLIDER, MAX_SLIDER].
+ *
+ * [range input]: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/range
+ *
+ * The returned Weight is exponential in the SliderPosition, so that e.g. if
+ * the SliderPosition is 2 then the Weight is 4, if the SliderPosition is 3
+ * then the Weight is 8, and so forth. The exception is when the
+ * SliderPosition==MIN_SLIDER_VALUE. In this case, the Weight will be 0.
+ */
+export type SliderPosition = number;
+
+export const MIN_SLIDER: SliderPosition = -5;
+export const MAX_SLIDER: SliderPosition = 5;
 
 export type Props = {|
-  +weight: number,
+  // The current Weight the WeightSlider is set to
+  +weight: Weight,
+  // The name associated with the slider.
+  // Will be displayed adjacent to the slider.
   +name: React$Node,
-  +onChange: (number) => void,
+  // onChange handler so the parent can recieve (and then set) the new weight
+  +onChange: (Weight) => void,
 |};
+
+/**
+ * The WeightSlider is a React component for letting the user input
```

Comment:
It would probably be helpful to explicitly note that this is a
[“controlled component”][cc].

[cc]: https://reactjs.org/docs/forms.html#controlled-components

Reply:
Done.